src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,67 @@
+package duke;
+
+import java.time.LocalDate;
+
+/**
+ * Takes in command from user and checks what to do with it
+ */
+public class Parser {
+    TaskList taskList = new TaskList();
+
+    Parser() {
+    }
+
+    /**
+     * Check if description is missing
+     *
+     * @param input
+     * @return validity
+     */
+    private boolean checkInvalidDescription(String input) {
+        if (input.trim().equals("todo") || input.trim().equals("deadline") || input.trim().equals("event")) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Take in input from user and checks what commands to run
+     *
+     * @param input from user
+     */
+    public void commands(String input) {
+        try {
+            if (input.equals("list")) {
+                taskList.printList();
+            } else if (input.length() > 5 && input.substring(0, 5).equals("done ")) {
+                String[] split = input.split("\\s+");
+                int number = Integer.parseInt(split[1]);
+                taskList.changeStatus(number);
+            } else if (input.length() > 9 && input.substring(0, 9).equals("deadline ")) { //date then time
+                String[] split = input.substring(9).split(" /by ");
```

Comment:
Maybe not entirely necessary but it might be good to use more descriptive array names here, to complement comments like the one directly below explaining what the splitting does.

Reply:
Good point!